### ISSUE TITLE
Uptobox doesn't detect real wait time

### ISSUE THREAD

**Issue by youlJunior** [pyload/pyload#1904]
When using Uptobox free, it doesn't detect real wait time (max 45 minutes), and only wait 60 seconds.

The reason is that the error string "You have to wait 16 minutes, 46 seconds to launch a new download" is not correctly detected.

It can be corrected by adding a pattern to match in SimpleHoster.py at line 384 :
orig : if re.search('limit|wait|slot', errmsg, re.I):
corrected : if re.search('limit|wait|slot|attendre', errmsg, re.I):

The problem is that the error string is localized so in my test the string is in French, so i added the word "attendre" in regex, because my error string was "Vous devez attendre 16 minutes, 46 seconds avant..."

**Comment by f6fvy** [comment]
A more generic way would be for pyload to fetch the english version of uptobox if possible.

**Comment by youlJunior** [comment]
If we declare ERROR_PATTERN in Uptobox hoster, does it override the pattern declared in XSFHoster ?

If so, we can make another regex which will catch second error message which is always in english.

**Comment by vuolter** [comment]
XFS based hosters (like Uptobox) should be always parsed in english, because pyload set their language cookie before.

**Comment by youlJunior** [comment]
So maybe there is a problem with the cookie, because the error string captured from the HTML, and written in the log file is in french...

**Comment by vuolter** [comment]
Could you post a sample debug log, please?

**Comment by youlJunior** [comment]
I can't provide you log now, because of bug #1913...

**Comment by vuolter** [comment]
Fixing...

**Comment by youlJunior** [comment]
OK, thks, but I saw that in XFSHoster : 

def set_xfs_cookie(self):
        if not self.COOKIES:
            return

        if isinstance(self.COOKIES, list) and (self.PLUGIN_DOMAIN, "lang", "english") not in self.COOKIES:
            self.COOKIES.insert((self.PLUGIN_DOMAIN, "lang", "english"))
        else:
            set_cookie(self.req.cj, self.PLUGIN_DOMAIN, "lang", "english")

The class XFSHoster doesn't have COOKIES atrribute, so it means that if the Uptobox class (which use XFSHoster) doesn't have COOKIES attribute, the function return, and no cookie is set.

I see two possibles fixes : 
- Create a COOKIE attribute in Uptobox Hoster class
- Rewrite the code like that : 
def set_xfs_cookie(self):
        if self.COOKIES:
            if isinstance(self.COOKIES, list) and (self.PLUGIN_DOMAIN, "lang", "english") not in self.COOKIES:
				self.COOKIES.insert((self.PLUGIN_DOMAIN, "lang", "english"))
        else:
            set_cookie(self.req.cj, self.PLUGIN_DOMAIN, "lang", "english")

**Comment by vuolter** [comment]
self.COOKIES It is inherited from the class SimpleHoster.

**Comment by youlJunior** [comment]
Mmmmm OK...

I tried another thing : I set a COOKIE in Firefox, in order to set english language, and tried to open a dl on uptobox.

In firefox, i the header of the uptobox page (where the is "Home" link) is now in english, but the erro message is still in FRENCH !!.

Here is a part of the html code : 
````
<br><p class="err"><div style="margin-top: 0px; margin-left: auto; margin-right: auto; width: 1200px;">
         <b><font size="2" color="red">Vous devez attendre 12 minutes, 46 seconds avant de pouvoir t&eacute;l&eacute;charger à nouveau</font><br /><br /><a href='http://uptobox.com/?op=registration&next=prem30' class="blue_link" style="font-size:18px;">Inscrivez vous </a> <font size="2" color="red">pour réduire votre temps d'attente ou </font><a href='http://uptobox.com/?op=registration&next=prem30' class="blue_link" style="font-size:18px;">devenez membre premium</a> <font size="2" color="red">pour supprimer toutes les limites!!!</font></b>
<br /><br /><b><font size="2" color="red">You have to wait 12 minutes, 46 seconds to launch a new download</font><br /><br /><a href='http://uptobox.com/?op=registration&next=prem15' class="blue_link" style="font-size:18px;">Register </a> <font size="2" color="red">to reduce the waiting time or </font><a href='http://uptobox.com/?op=registration&next=prem15' class="blue_link" style="font-size:18px;">become a premium member</a> <font size="2" color="red">to bypass all limits!!!</font></b><br /><br /><iframe src="http://uptobox.com/utb_728_wait.php" frameborder="0" width="735" height="180" scrolling="no" name="counter"></iframe><br /></center><!--<iframe src="http://cometodress.com" frameborder="0" width="0" height="0" scrolling="no" name="counter"></iframe>--><!--<script type="text/javascript">
````

**Comment by youlJunior** [comment]
Now you fixed the #1913, here is the log : 

01.10.2015 23:26:05	INFO	HOSTER UptoboxCom[146]: Waiting 60 seconds...
171	01.10.2015 23:26:05	DEBUG	HOSTER UptoboxCom[146]: RECONNECT not required | Previous wantReconnect: True
170	01.10.2015 23:26:05	DEBUG	HOSTER UptoboxCom[146]: WAIT set to 60 seconds | Previous waitUntil: 0.000000
169	01.10.2015 23:26:05	WARNING	HOSTER UptoboxCom[146]: Vous devez attendre 1 minute, 37 seconds avant de pouvoir télécharger à nouveau
168	01.10.2015 23:26:05	DEBUG	HOSTER UptoboxCom[146]: Getting download link #1
167	01.10.2015 23:26:05	INFO	HOSTER UptoboxCom[146]: Processing as free download...
166	01.10.2015 23:26:04	DEBUG	HOSTER UptoboxCom[146]: LOAD URL http://uptobox.com/1w1f3fsxih37 | cookies=True | get={} | req=None | decode=True | multipart=False | post={} | ref=False | just_header=False
165	01.10.2015 23:26:04	INFO	HOSTER UptoboxCom[146]: File status: démarrage
164	01.10.2015 23:26:04	INFO	HOSTER UptoboxCom[146]: File size: 432642457 bytes
163	01.10.2015 23:26:04	INFO	HOSTER UptoboxCom[146]: File name: Defiance.S03E10.FRENCH.WEB-DL.x264-LiBERTY.mp4
162	01.10.2015 23:26:04	DEBUG	HOSTER UptoboxCom[146]: Previous file info: {}
161	01.10.2015 23:26:04	DEBUG	HOSTER UptoboxCom[146]: File info: {'status': 2, 'url': u'http://uptobox.com/1w1f3fsxih37', 'size': 432642457.6, 'name': u'Defiance.S03E10.FRENCH.WEB-DL.x264-LiBERTY.mp4', 'pattern': {'S': u'412.6', 'U': u'MB', 'N': u'Defiance.S03E10.FRENCH.WEB-DL.x264-LiBERTY.mp4'}}
160	01.10.2015 23:26:04	INFO	HOSTER UptoboxCom[146]: Updating file info...
159	01.10.2015 23:26:04	DEBUG	HOSTER UptoboxCom[146]: PROCESS URL http://uptobox.com/1w1f3fsxih37 | PLUGIN VERSION 0.21
158	01.10.2015 23:26:04	DEBUG	HOSTER UptoboxCom[146]: Config option `use_premium` not found, use default `True`
157	01.10.2015 23:26:04	DEBUG	ADDON UserAgentSwitcher: Use custom user-agent string: Mozilla/5.0 (Windows NT 6.1; WOW64; rv:38.0) Gecko/20100101 Firefox/38.0
156	01.10.2015 23:26:04	INFO	Download starts: Defiance.S03E10.FRENCH.WEB-DL.x264-LiBERTY.mp4
155	01.10.2015 23:26:04	DEBUG	Redirected import module.plugins.internal.Account -> userplugins.internal.Account
154	01.10.2015 23:26:04	DEBUG	Redirected import module.plugins.internal.XFSAccount -> userplugins.internal.XFSAccount
153	01.10.2015 23:26:04	DEBUG	Redirected import module.plugins.accounts.UptoboxCom -> userplugins.accounts.UptoboxCom
152	01.10.2015 23:26:04	DEBUG	HOSTER UptoboxCom[146]: Config option `use_premium` not found, use default `True`
151	01.10.2015 23:26:04	DEBUG	Redirected import module.plugins.internal.Base -> userplugins.internal.Base
150	01.10.2015 23:26:04	DEBUG	Redirected import module.plugins.internal.Hoster -> userplugins.internal.Hoster
149	01.10.2015 23:26:04	DEBUG	Redirected import module.plugins.internal.SimpleHoster -> userplugins.internal.SimpleHoster
148	01.10.2015 23:26:04	DEBUG	Redirected import module.plugins.internal.Captcha -> userplugins.internal.Captcha
147	01.10.2015 23:26:04	DEBUG	Redirected import module.plugins.internal.XFSHoster -> userplugins.internal.XFSHoster
